Formula/alass.rb

```diff
@@ -0,0 +1,20 @@
+class Alass < Formula
+  version "v2.0.0"
+  sha256 "ce88f92c7a427b623edcabb1b64e80be70cca2777f3da4b96702820a6cdf1e26"
+  
+  url "https://github.com/kaegi/alass/archive/refs/tags/#{version}.tar.gz"
+  desc "Automatic Language-Agnostic Subtitle Synchronization"
+  homepage "https://github.com/kaegi/alass"
+  license "GPL-3.0-or-later"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "build", "--release", "--bin", "alass-cli"
+    bin.install "target/release/alass-cli"
+  end
+
+  test do
+    system "#{bin}/alass-cli", "--help"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thank you for the feedback - I'm trying to find a test case that makes sense 👍
Thing is: we need a video file with subtiles to test the application -> I don't know how to deal with that from a licensing standpoint.

Reply:
> If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.

This can also apply to "If the software cannot function without certain files"

Reply:
Could you maybe help me?
I'm not quite able how to insert the multiline output.
My current test:
```rb
  test do
    # create test files
    (testpath/"reference.srt").write <<~EOS
        1
        00:00:00,000 --> 00:00:01,000
        This is the first subtitle.
    EOS
    (testpath/"incorrect.srt").write <<~EOS
        1
        00:00:01,000 --> 00:00:02,000
        This is the first subtitle.
    EOS

    # run test
    output = shell_output("#{bin}/alass-cli reference.srt incorrect.srt output.srt").strip
    assert_equal "shifted block of 1 subtitles with length 0:00:00.000 by -0:00:01.000", output
  end
  ```
  
  The expected output is:
  ```sh
  Guessing framerate ratio...
6 / 6 [========================================================================================================================================] 100.00 % 41896.81/s 

info: 'reference file FPS/input file FPS' ratio is 1

synchronizing 'incorrect.srt' to reference file 'reference.srt'...
1 / 1 [=======================================================================================================================================] 100.00 % 198333.99/s 

shifted block of 1 subtitles with length 0:00:00.000 by -0:00:01.000
```

Thing is: we can't predict some of the output since it's displaying the speed...

Reply:
You don't need to match everything, and `assert_match` also allows a regex if needed

Reply:
Thank you for the help - I got it 👍